Mathlib/Tactic/PushNeg.lean

```diff
@@ -55,7 +55,7 @@ def transformNegationStep (e : Expr) : SimpM (Option Simp.Step) := do
     catch _ => return none
   let e_whnf ← whnfR e
   let some ex := e_whnf.not? | return Simp.Step.visit { expr := e }
-  match ex.getAppFnArgs with
+  match ex.cleanupAnnotations.getAppFnArgs with
```

Comment:
Oh, I made two mistakes with my suggestion. First, `ex` is used later in the default case of the `match`, and second there might still be metavariables in `e`.

Here's a change that will also make it behave somewhat more like the `rfl`/`symm`/`trans` tactics.  We don't need this sort of solution on line 56 since `whnfR` will strip metadata and instantiate just enough metavariables for line 57 to work.

```suggestion
  let ex := (← instantiateMVars ex).cleanupAnnotations
  match ex.getAppFnArgs with
```

Reply:
I pushed this change, thanks!

I do not really understand the deep reasons for it, but I appreciate very much the comments!